package ba.test;
// Generated by ComTest BEGIN
import static org.junit.Assert.*;
import org.junit.*;
import ba.*;
// Generated by ComTest END

/**
 * Test class made by ComTest
 * @version 2022.03.01 22:47:25 // Generated by ComTest
 *
 */
@SuppressWarnings({ "all" })
public class BaTest {



  // Generated by ComTest BEGIN
  /** 
   * testAdd35 
   * @throws TilaException when error
   */
  @Test
  public void testAdd35() throws TilaException {    // Ba: 35
    Ba ba = new Ba(); 
    Area a1 = new Area(), a2 = new Area(); 
    a1.register(); a2.register(); 
    assertEquals("From: Ba line: 40", 0, ba.getAreaCount()); 
    ba.add(a1); assertEquals("From: Ba line: 41", 1, ba.getAreaCount()); 
    ba.add(a2); assertEquals("From: Ba line: 42", 2, ba.getAreaCount()); 
    ba.add(a1); assertEquals("From: Ba line: 43", 3, ba.getAreaCount()); 
    assertEquals("From: Ba line: 44", a1, ba.getArea(0)); 
    assertEquals("From: Ba line: 45", a2, ba.getArea(1)); 
    assertEquals("From: Ba line: 46", a1, ba.getArea(2)); 
    try {
    assertEquals("From: Ba line: 47", a1, ba.getArea(3)); 
    fail("Ba: 47 Did not throw IndexOutOfBoundsException");
    } catch(IndexOutOfBoundsException _e_){ _e_.getMessage(); }
    ba.add(a1); assertEquals("From: Ba line: 48", 4, ba.getAreaCount()); 
    ba.add(a1); assertEquals("From: Ba line: 49", 5, ba.getAreaCount()); 
    try {
    ba.add(a1); 
    fail("Ba: 50 Did not throw TilaException");
    } catch(TilaException _e_){ _e_.getMessage(); }
  } // Generated by ComTest END
}